### PR TITLE
Fix scriptslog multiply window

### PR DESCRIPTION
**Pull request by vladest**
https://highfidelity.fogbugz.com/f/filters/177/My-Cases

Test plan:
- Open Developer > Script Log (HMD Friendly) several times.  
- make sure only one window created
- Close and reopen the window. 
- Make sure it still works

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4353/HighFidelity-Beta-PR11653-3070b44c90241fd4567b5fb249ac739ba516dfb8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4353/HighFidelity-Beta-PR11653-3070b44c90241fd4567b5fb249ac739ba516dfb8.exe

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -7311,6 +7317,10 @@ void Application::switchDisplayMode() {
     _previousHMDWornStatus = currentHMDWornStatus;
 }
 
+void Application::cleanupRunningScripts(const QString &fileNameString, ScriptEnginePointer) {
```
Comment (ZappoMan) [inline interface/src/Application.cpp]: coding standard: `const QString &fileNameString` should be `const QString& fileNameString`

**Comment by vladest** [comment]
@ZappoMan for future usage

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4388/HighFidelity-Beta-PR11653-c55f93813237d2a3010237b0e5dcc6ded12f1f82.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4388/HighFidelity-Beta-PR11653-c55f93813237d2a3010237b0e5dcc6ded12f1f82.exe

**Comment by vladest** [comment]
@ZappoMan what is your proposals? boolean for the log window only?

**Comment by ZappoMan** [comment]
All of this can be done inside of the script itself... we don't need to add any new C++ code.

Instead the script can check that it's already running by calling ScriptEngines.getRunning() and then looking for itself in the list... and if so... stop the second instance.

**Comment by vladest** [comment]
@ZappoMan just checked: when script running its already contains its name in getRunning() output

**Comment by YuppyPlays** [comment]
Awaiting build for QA.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4756/HighFidelity-Beta-PR11653-5e32298183caa1cc7ef8f1a05d372d545cf6bef8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4756/HighFidelity-Beta-PR11653-5e32298183caa1cc7ef8f1a05d372d545cf6bef8.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

Open Developer > Script Log (HMD Friendly) several times. **Check**
make sure only one window created **Check**
Close and reopen the window. **Check, though I managed to break it by pressing the HMD friendly script button 5 times or more, then closing the actual window, after this you cannot seem to re-open the script window.
Make sure it still works **Check, except for the previous mentioned issue**

**Fail** Unless the previously found break/issue is non-PR related.

**Comment by Nex-Pro** [comment]
### FAIL this is what is happening.

205564-hifi-snap-by--on-2017-11-08_19-34-49

When the window is lost it can be restored with ctrl+r

However the script isn't being cleaned up on re-run.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4773/HighFidelity-Beta-PR11653-02625a6e28d63c0f6fb7f488eb3b1d310a3d8703.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4773/HighFidelity-Beta-PR11653-02625a6e28d63c0f6fb7f488eb3b1d310a3d8703.dmg

**Comment by Nex-Pro** [comment]
Testing

**Comment by Nex-Pro** [comment]
@vladest 

### PASS

All looks good. I observed the running scripts list and no additional copies of the script appeared there. Can close log and restart it from the menu with no problems. Can also load from folder list in running scripts.  Attempting to reload the log window from the menu will put these messages in the log window.

[10/8 23:56:5] [debugWindow.js] INFO - Script Engine starting:debugWindow.js
[10/8 23:56:5] [debugWindow.js] INFO - Script Engine stopping:debugWindow.js

Hitting the restart icon in running scripts resarts the log window and clears it.

Log window persists across restarts if running when Interface is closed.

***Here is the log output from multiple attempts to run the script from the menu, observe there is only one entry for debugWindow.js in the list.***

f11068-hifi-snap-by--on-2017-11-08_23-59-02

**Pull request merged**